src/twisted/web/test/test_flatten.py

```diff
@@ -706,15 +706,15 @@ def render(self, request: Optional[IRequest]) -> Flattenable:
                     Exception while flattening:
                       \\[<unrenderable>\\]
                       <unrenderable>
-                      .*
+                      <Deferred at .* current result: <twisted.python.failure.Failure builtins.RuntimeError: example>>
                       File ".*", line \\d*, in _flattenTree
-                        element = await element
-                    RuntimeError: example
+                        element = await element.*
                     """
                 ),
                 flags=re.MULTILINE,
             ),
         )
+        self.assertIn("RuntimeError: example", str(failure.value))
```

Comment:
Just one last call

To me, the previous assertion `assertRegex` already does what the `assertIn` does.

And `assertIn` doesn't assert the order  or the place where the error is raised.

```
self.assertRegex(
    str(failure.value),
    'BLA<twisted.python.failure.Failure builtins.RuntimeError: example>>BLA',
    )
```

Reply:
I agree that, ideally, that's what we'd do. But as @eevelweezel put it above:

> I still couldn't get this to match

and I think we should be mindful of the amount of effort it takes to arrive at an ideal solution.

If anyone wants to submit a cleanup in trunk after this is landed I think that would be great.